src/main/java/gift/member/MemberService.java

```diff
@@ -25,26 +23,32 @@ public MemberService(
     }
 
     public String register(MemberDTO memberDTO) {
-        if (memberRepository.existsById(memberDTO.getEmail())) {
-            throw new IllegalArgumentException(MEMBER_ALREADY_EXISTS);
-        }
-
-        memberRepository.save(Member.fromMemberDTO(memberDTO));
-
-        return jwtProvider.generateToken(MemberTokenDTO.fromMemberDTO(memberDTO));
+        memberRepository.findById(memberDTO.getEmail())
+            .ifPresentOrElse(
+                e -> {
+                    throw new IllegalArgumentException(MEMBER_ALREADY_EXISTS);
+                },
+                () -> memberRepository.save(memberDTO.toEntity())
+            );
+
+        return jwtProvider.generateToken(memberDTO.toTokenDTO());
     }
 
     public String login(MemberDTO memberDTO) {
-        Optional<Member> findMember = memberRepository.findById(memberDTO.getEmail());
-
-        if (findMember.isEmpty()) {
-            throw new FailedLoginException(MEMBER_NOT_FOUND);
-        }
+        memberRepository.findById(memberDTO.getEmail())
+            .ifPresentOrElse(
+                findMember -> verifyPassword(findMember, memberDTO),
+                () -> {
+                    throw new FailedLoginException(MEMBER_NOT_FOUND);
+                }
+            );
```

Comment:
ifPresentOrElse 사용 👍 
findBy000은 보통 orElseThrow와 함께 자주 쓰인답니다 😄 
참고로 알아두면 좋아요!

```suggestion
        Member findMember = memberRepository.findById(memberDTO.getEmail())
                .orElseThrow(() -> new FailedLoginException(MEMBER_NOT_FOUND));
        verifyPassword(findMember, memberDTO);
```